odoo/addons/base/models/res_users.py

```diff
@@ -654,6 +654,18 @@ def _check_one_user_type(self):
             if self._has_multiple_groups(user_types_groups.ids):
                 raise ValidationError(_('The user cannot have more than one user types.'))
 
+    @api.constrains("login")
+    def _check_login(self):
```

Comment:
Why is this needed? We alrady have a unique index on `login`. (check attribute `_login_key`).
Also the implementation checks the rowcount for the last execute.